src/editmap.cpp

```diff
@@ -1831,6 +1831,170 @@ int editmap::mapgen_preview( real_coords &tc, uimenu &gmenu )
+        if ( dist > r || dist == 0 ) {
+            popup( _("You must select a tile within %d range of the camp"), r );
+            return false;
+        }
+
+        omt_tgt = tripoint(where.x, where.y, omt_tgt.z);
+        oter_id &omt_test = overmap_buffer.ter( omt_tgt.x, omt_tgt.y, omt_tgt.z );
+        if( omt_test.id() != "field" ){
+            popup( _("You must construct expansions in fields.") );
+            return false;
+        }
+    }
+
+    // Coordinates of the overmap terrain that should be generated.
+    oter_id &omt_ref = overmap_buffer.ter( omt_tgt.x, omt_tgt.y, omt_tgt.z );
+    omt_ref = oter_id(om_name);
+
+    tinymap target_bay;
+    target_bay.load( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, false );
+
+    tinymap tmpmap;
+    tmpmap.generate( omt_tgt.x * 2, omt_tgt.y * 2, target.z, calendar::turn );
+    point target_sub( target.x / 12, target.y / 12 );
+
+    g->m.clear_vehicle_cache( target.z );
+    for( int x = 0; x < 2; x++ ) {
+        for( int y = 0; y < 2; y++ ) {
+            submap *destsm = target_bay.get_submap_at_grid( x, y, target.z );
+            submap *srcsm = tmpmap.get_submap_at_grid( x, y, target.z );
+            destsm->is_uniform = false;
+            srcsm->is_uniform = false;
+
+            if( !destsm->vehicles.empty() ){
+                popup( _("Engine cannot support merging vehicles from two overmaps, please remove them from the OM tile.") );
+                return false;
+            }
+
+            destsm->delete_vehicles();
+            for( size_t i = 0; i < srcsm->vehicles.size(); i++ ) { // copy vehicles to real map
+                vehicle *veh1 = srcsm->vehicles[i];
+                veh1->smx = target_sub.x + x;
+                veh1->smy = target_sub.y + y;
+                veh1->smz = target.z;
+                destsm->vehicles.push_back( veh1 );
+                g->m.update_vehicle_cache( veh1, target.z );
+            }
+            srcsm->vehicles.clear();
+            g->m.update_vehicle_list( destsm, target.z );
+
+            int spawns_todo = 0;
+            for( size_t i = 0; i < srcsm->spawns.size(); i++ ) { // copy spawns
+                destsm->spawns.push_back( srcsm->spawns[i] );
+                spawns_todo++;
+            }
+
+            destsm->field_count = srcsm->field_count; // and count
+            std::memcpy( destsm->trp, srcsm->trp, sizeof( srcsm->trp ) ); // traps
+            std::memcpy( destsm->rad, srcsm->rad, sizeof( srcsm->rad ) ); // radiation
+            std::memcpy( destsm->lum, srcsm->lum, sizeof( srcsm->lum ) ); // emissive items
+
+
+            for( int sx = 0; sx < SEEX; ++sx ) {
+                for( int sy = 0; sy < SEEY; ++sy ) {
+                    for( auto &elem : srcsm->itm[sx][sy] ) {
+                        destsm->itm[sx][sy].push_back(elem);
+                    }
+                    //Don't cover existing crops, for farm upgrades
+                    //Don't destroy terrain to place grass/dirt
```

Comment:
You want `mapgendata::is_groundcover` here so that it works in the desert too.

Reply:
It's in `mapgendata`, but doesn't actually depend on it since it only checks data copied from `regional_settings`.
It's declared and defined in `mapgen_functions.h/.cpp`. Moving/adding it to `regional_settings` would be easy and useful here.

Reply:
Just noticed, that is actually coming from the player defined json.  After a quick update, all grass and dirt tiles in the user defined json will be ignored unless they have furniture on them also.  Not looking at the world that is loading the json  like with the plants above it.  I tried to clear it up a little.